### PR TITLE
Singup

### PR DESCRIPTION
**Pull request by xavierloos**
As a user
So that I can join AbodenB
I would like to sign-up

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      newuser = instance_double(User)
+      allow(newuser).to receive(:username).and_return("test")
+      allow(newuser).to receive(:email).and_return("test@test.com")
+      allow(newuser).to receive(:password).and_return("test")
+      expect(newuser.username).to eq "test"
+      expect(newuser.email).to eq "test@test.com"
+      expect(newuser.password).to eq "test"
+    end
+  end
+
```
Comment (ollienorman) [inline spec/user_spec.rb:8]: First test appears to be a false positive - it's creating a mock (in lines 6 to 9) and the testing that mock (in lines 10 to 12). It doesn't actually use the user.rb code

File: spec/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  describe '.create' do
+    it 'creates a new user' do
+      user = User.create(username: 'test_user', email: 'test@test.com', password: 'test_password')
+      persisted_data = persistent_data(table: 'users', id: user.id)
+      p user.id
```
Comment (Will-Helliwell) [inline spec/user_spec.rb]: p statements can be deleted

File: app.rb [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+require "pg"
 class AbodenB < Sinatra::Base
   get "/" do
     "Welcome"
   end
+  get "/signup" do
+    erb :signup
+  end
+  post "/signup" do
+    redirect "/profile"
```
Comment (ollienorman) [inline app.rb:19]: Not calling the User.create method here - when feature testing the website manually, it isn't adding the signup details to the database

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      allow(newuser).to receive(:password).and_return("test")
+      expect(newuser.username).to eq "test"
+      expect(newuser.email).to eq "test@test.com"
+      expect(newuser.password).to eq "test"
+    end
+  end
+
+  describe '.create' do
+    it 'creates a new user' do
+      user = User.create(username: 'test_user', email: 'test@test.com', password: 'test_password')
```
Comment (ollienorman) [inline spec/user_spec.rb]: Should probably add a setup_test_database helper to truncate the database before each test

**Review by Will-Helliwell — COMMENTED** [review]
Also a couple of debugging p statements that can be removed

File: spec/user_spec.rb (hunk 2 above)
Comment (xavierloos) [reply spec/user_spec.rb]: done!

File: spec/user_spec.rb (hunk 1 above)
Comment (calavell) [reply spec/user_spec.rb:8]: this test is now redundant so we have removed it

File: app.rb (hunk 3 above)
Comment (calavell) [reply app.rb:19]: resolved

File: spec/user_spec.rb (hunk 4 above)
Comment (calavell) [reply spec/user_spec.rb]: will do this afternoon

**Review by Will-Helliwell — APPROVED** [review]
Great stuff! :shipit:

**Pull request merged**